autoload/test/zig/zigtest.vim

```diff
@@ -0,0 +1,48 @@
+if !exists('g:test#zig#zigtest#file_pattern')
+  let g:test#zig#zigtest#file_pattern = '\v\.zig$'
+endif
+
+if !exists('g:test#zig#zigtest#test_patterns')
+  let g:test#zig#zigtest#test_patterns = {
+        \ 'test': [
+        \ '\vtest "=([^"]*)"= \{',
+        \ ],
+        \ 'namespace': []
+        \ }
+endif
+
+" Returns true if the given file belongs to your test runner
+function! test#zig#zigtest#test_file(file) abort
+  return a:file =~# g:test#zig#zigtest#file_pattern
+endfunction
+
+" Returns test runner's arguments which will run the current file and/or line
+function! test#zig#zigtest#build_position(type, position) abort
+  if a:type ==# 'nearest'
+    let name = test#base#nearest_test(a:position, g:test#zig#zigtest#test_patterns)
+
+    if empty(name['test'])
+      return []
+    else
+      return ['test', a:position['file'], '--test-filter', ''''.name['test'][0].'''']
```

Comment:
> I'm new to both vimscript and zig, so I mostly copied this from the `bats` implementation. From what I can gather, [there isn't a way to use a pattern to run a single test with `zig test`, so there's a chance multiple tests match](https://github.com/ziglang/zig/issues/13524), but this seems close enough for now?

If that's the case, I'd suggest changing nearest to behave like testing a file, and comment the code accordingly

```vim
if a:type ==# 'nearest'
    return ['test', a:position['file']]
```